solutionbox/code_free_ml/analyze_data.py

```diff
@@ -0,0 +1,823 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import collections
+import csv
+import json
+import os
+import sys
+import pandas as pd
+import six
+import tensorflow as tf
+import tensorflow_transform as tft
+
+from tensorflow.contrib import lookup
+from tensorflow.python.lib.io import file_io
+from tensorflow_transform import impl_helper
+from tensorflow_transform.tf_metadata import dataset_metadata
+from tensorflow_transform.tf_metadata import dataset_schema
+from tensorflow_transform.tf_metadata import metadata_io
+
+# Files
+SCHEMA_FILE = 'schema.json'
+FEATURES_FILE = 'features.json'
```

Comment:
I originally called this file transforms (and is still called transforms internally) in the original SD package. However, Nikhil wanted this file called feature in the public api. So the current SD package names it a features file. I don't have strong opinions on this name, but we should keep the name to be comparable with SD.